WireDomain/Sources/WireDomain/Synchronization/SyncManager.swift

```diff
@@ -32,20 +38,103 @@ protocol SyncManagerProtocol {
 
 }
 
+public struct MLSProvider {
+    let service: any MLSServiceInterface
+    let isMLSEnabled: Bool
+
+    public init(
+        service: any MLSServiceInterface,
+        isMLSEnabled: Bool
+    ) {
+        self.service = service
+        self.isMLSEnabled = isMLSEnabled
+    }
+}
+
 final class SyncManager: SyncManagerProtocol {
 
+    enum Error: Swift.Error {
+        case failedToPerformSlowSync(Swift.Error)
+    }
```

Comment:
Perhaps `SyncManagerError`, this way we avoid having to fully qualify `Swift.Error`?

Reply:
Since the enum is nested inside the SyncManager, we know the error is a SyncManager error, also it feels more intuitive at callsite: `SyncManager.Error` rather than `SyncManager.SyncManagerError` which seems redundant. It is also aligned with what we did for processors. 

On a more general perspective, I would go with either of the two following approaches:

1/ Call it _NameOfTheObject_**Error** but don't nest it inside the object
2/ Call it Error and have it nested inside the object.

Both solutions work for me, in WireDomain we use both approaches though, we should streamline this to one or the other solution.

Reply:
done in https://github.com/wireapp/wire-ios/pull/1999/commits/e151fbc4913b20e97bb8214302a8f3558944479c